app/cdap/components/Connections/Create/index.tsx

```diff
@@ -180,7 +180,20 @@ export function CreateConnection({
       }
 
       if (connectionMode === IConnectionMode.ROUTED && enableRouting) {
-        setRedirectUrl(`${getConnectionPath(name)}`);
+        /**
+         * This following code is checking whether the add connection request is coming from New UI
+         * If the request is from New UI, the redirection is set to New UI URL
+         * Otherwise, the redirection is set to Old UI URL as existed
+         */
+        const value: string = localStorage.getItem('addConnectionRequestFromNewUI');
+        if (value) {
+          localStorage.removeItem('addConnectionRequestFromNewUI');
```

Comment:
so you're using local storage as state management/global variable? I think this would be better as a query parameter or something like that. @njbriggs what do you think? I think generally local storage shouldn't be used to do app state management.

Reply:
Removed local storage and handled the re-direction with the help of react-router's useLocation hook.